ideascube/serveradmin/tests/test_models.py

```diff
@@ -0,0 +1,45 @@
+from datetime import datetime, timezone
+
+import freezegun
+import pytest
+
+
+@pytest.mark.parametrize(
+    'value, type_method',
+    [
+        (True, 'as_boolean'),
+        (42, 'as_integer'),
+        (None, None),
+        ('A string', 'as_string'),
+    ],
+    ids=[
+        'boolean',
+        'int',
+        'none',
+        'string',
+    ])
+@pytest.mark.usefixtures('db')
+def test_settings(value, type_method, user):
+    from ideascube.serveradmin.models import Setting
+
+    fakenow = datetime.now(tz=timezone.utc)
+    assert Setting.objects.count() == 0
+
+    with freezegun.freeze_time(fakenow):
+        Setting(
+            namespace='tests', key='setting1', value=value, actor=user).save()
+
+    assert Setting.objects.count() == 1
+
+    setting = Setting.objects.first()
+    assert setting.namespace == 'tests'
+    assert setting.key == 'setting1'
+    assert setting.value == value
+    assert setting.actor == user
+    assert setting.date == fakenow
+    assert str(setting) == 'tests.setting1=%r' % value
+
+    if type_method is not None:
+        for method in filter(lambda m: m.startswith('as_'), dir(setting)):
```

Comment:
Well yeah, all settings have all type methods.

So to avoid looping, the solution is:

```python
assert getattr(setting, type_method)() == value
```

The looping is indeed entirely unnecessary here, I had it because at first I was testing that calling the corresponding `type_method` would succeed **and** calling the other methods would raise (see the other discussion about how I went from checking `isinstance` to casting).

But when the methods stopped raising, I removed the calls to the other `as_methods`, and left that loop, without realizing it was entirely pointless.

But in any case, this will completely change as I redo the API.